runtime/runtime/src/lib.rs

```diff
@@ -1246,6 +1271,36 @@ impl Runtime {
 
         state_update.commit(StateChangeCause::UpdatedDelayedReceipts);
 
+        checked_feature!(
+            "protocol_feature_transaction_hashes_in_state",
+            TransactionHashesInState,
+            apply_state.current_protocol_version,
+            {
+                set(
+                    &mut state_update,
+                    TrieKey::BlockTransactionHashes { height: apply_state.block_index },
+                    &transaction_hash_keys,
+                );
+                for height in apply_state
+                    .prev_block_height
+                    .saturating_sub(apply_state.transaction_validity_period)
+                    ..apply_state
+                        .block_index
+                        .saturating_sub(apply_state.transaction_validity_period)
```

Comment:
Tiny nit, but feels like this can benefit from introduction of a couple of local variables.

Reply:
It would be great to comment and draw an example with the boundaries of valid TXs and when they expire the most. Include skipped block in this example as well.

I did it on a paper and it seems fine, but we should include it here for future reference.